crates/matrix-sdk-base/src/sliding_sync.rs

```diff
@@ -767,6 +792,196 @@ mod test {
+    }
+
+    #[test]
+    fn only_store_the_max_number_of_encrypted_events() {
+        // Given two message events followed by lots of encrypted and other irrelevant
+        // events
+        let evente = make_event("m.room.message", "$e");
+        let eventd = make_event("m.room.message", "$d");
+        let eventc = make_encrypted_event("$c");
+        let event9 = make_encrypted_event("$9");
+        let event8 = make_encrypted_event("$8");
+        let event7 = make_encrypted_event("$7");
+        let eventb = make_event("m.read", "$b");
+        let event6 = make_encrypted_event("$6");
+        let event5 = make_encrypted_event("$5");
+        let event4 = make_encrypted_event("$4");
+        let event3 = make_encrypted_event("$3");
+        let event2 = make_encrypted_event("$2");
+        let eventa = make_event("m.read", "$a");
+        let event1 = make_encrypted_event("$1");
+        let event0 = make_encrypted_event("$0");
+        let events = &[
+            evente.clone(),
+            eventd.clone(),
+            eventc.clone(),
+            event9.clone(),
+            event8.clone(),
+            event7.clone(),
+            eventb.clone(),
+            event6.clone(),
+            event5.clone(),
+            event4.clone(),
+            event3.clone(),
+            event2.clone(),
+            eventa.clone(),
+            event1.clone(),
+            event0.clone(),
+        ];
+
+        // When I ask to cache events
+        let mut room_info = RoomInfo::new(room_id!("!r:h.uk"), RoomState::Joined);
+        cache_latest_events(&mut room_info, events);
+
+        // The latest message is stored, ignoring encrypted and receipts
+        assert_eq!(ev_id(room_info.latest_event), rawev_id(eventd));
+
+        // Only 10 encrypted are stored, even though there were more
+        assert_eq!(
+            rawevs_ids(&room_info.latest_encrypted_events),
+            evs_ids(&[
+                event9, event8, event7, event6, event5, event4, event3, event2, event1, event0
+            ])
+        );
+    }
+
+    #[test]
+    fn dont_overflow_capacity_if_previous_encrypted_events_exist() {
+        // Given a RoomInfo with lots of encrypted events already inside it
+        let mut room_info = RoomInfo::new(room_id!("!r:h.uk"), RoomState::Joined);
+        cache_latest_events(
+            &mut room_info,
+            &[
+                make_encrypted_event("$0"),
+                make_encrypted_event("$1"),
+                make_encrypted_event("$2"),
+                make_encrypted_event("$3"),
+                make_encrypted_event("$4"),
+                make_encrypted_event("$5"),
+                make_encrypted_event("$6"),
+                make_encrypted_event("$7"),
+                make_encrypted_event("$8"),
+                make_encrypted_event("$9"),
+            ],
+        );
+        // Sanity: room_info has 10 encrypted events inside it
+        assert_eq!(room_info.latest_encrypted_events.len(), 10);
```

Comment:
Can't we use `RoomInfo::MAX_ENCRYPTED_EVENTS` instead of 10 here? If it's not accessible, then fine, let's keep 10 :-).

Reply:
It's not accessible at the moment, and I sometimes kinda like being very explicit in tests, so might even argue to leave this as-is, so resolving this for now, but feel free to re-open.